kiwi/bootloader/config/systemd_boot.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) 2022 Marcus Schäfer
+#
+# This file is part of kiwi.
+#
+# kiwi is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# kiwi is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with kiwi.  If not, see <http://www.gnu.org/licenses/>
+#
+import os
+from string import Template
+from typing import Dict
+
+# project
+from kiwi.path import Path
+from kiwi.bootloader.template.systemd_boot import BootLoaderTemplateSystemdBoot
+from kiwi.bootloader.config.bootloader_spec_base import BootLoaderSpecBase
+from kiwi.command import Command
+
+from kiwi.exceptions import (
+    KiwiTemplateError,
+    KiwiBootLoaderTargetError
+)
+
+
+class BootLoaderSystemdBoot(BootLoaderSpecBase):
+    def create_loader_image(self, target: str) -> None:
+        self.efi_boot_path = self.create_efi_path()
+
+    def setup_loader(self, target: str) -> None:
+        if target != self.target.disk:
+            raise KiwiBootLoaderTargetError(
+                'systemd-boot is only supported with the EFI disk image target'
+            )
+        boot_options = self.custom_args['boot_options']
+        self._mount_system(
+            boot_options.get('root_device'),
+            boot_options.get('boot_device'),
+            boot_options.get('efi_device'),
+            boot_options.get('system_volumes')
+        )
+        BootLoaderSystemdBoot._write_config_file(
+            BootLoaderTemplateSystemdBoot().get_loader_template(),
+            self.root_mount.mountpoint + '/boot/efi/loader/loader.conf',
+            self._get_template_parameters()
+        )
+        Command.run(
+            ['chroot', self.root_mount.mountpoint, 'systemd-machine-id-setup']
+        )
+        Command.run(
+            [
+                'chroot', self.root_mount.mountpoint,
+                'bootctl', 'install', '--esp-path=/boot/efi', '--no-variables'
+            ]
+        )
+        # FIXME: this is strange, XBOOTLDR does not work for me
+        # with --boot=/boot at bootctl. How to make this work with a /boot
+        # not on fat, systemd-boot not able to read
+        # from a linux filesystem ?
+        Command.run(
+            [
+                'chroot', self.root_mount.mountpoint,
+                'cp', '-L', '/boot/vmlinuz', '/boot/efi/EFI/Linux'
+            ]
+        )
+        Command.run(
+            [
+                'chroot', self.root_mount.mountpoint,
+                'cp', '-L', '/boot/initrd', '/boot/efi/EFI/Linux'
+            ]
+        )
```

Comment:
As we discussed in the meeting today, we should drop this and verify that things work on Fedora (which is known to have working sd-boot support). @DaanDeMeyer is willing to help us out with questions and feedback as we iterate through this.

Reply:
yep, I deleted that code part and replaced it by the use of ```kernel-install```. With Fedora systems we now have a proper  support of systemd-boot.